gallery/visualtests/nxToast.js

```diff
@@ -0,0 +1,131 @@
+    const {
+      checkFullPageScreenshot,
+      getPage,
+      wait,
+      waitAndGetElements,
+      a11yTest
+    } = setupBrowser('#/NxToastComplexLayoutExample', false);
+
+    async function launchToasts(toastsContainer, buttonSelector) {
+
+      const [openToastBtn] = await waitAndGetElements(buttonSelector);
+      await openToastBtn.click();
+      await openToastBtn.click();
+
+      // wait for animation
+      await wait(500);
+
+      const [container] = await waitAndGetElements(toastsContainer);
+      return container;
+    }
+
+    describe('NxToastContainer', function() {
+
+      beforeEach(async function() {
+        await getPage().setViewport({ width: 1366, height: 1000 });
+      });
+
+      it('is positioned correctly', async function() {
+        await launchToasts(toastContainer, openButton);
+        await checkFullPageScreenshot();
+      });
+
+      it('passes a11y checks', async function() {
+        await launchToasts(toastContainer, openButton);
+        a11yTest();
+      });
+    });
+  });
+
+  describe('with global sidebar and without header', function() {
+    const {
+      checkFullPageScreenshot,
+      getPage,
+      wait,
+      waitAndGetElements,
+      a11yTest
+    } = setupBrowser('#/NxToastSimpleLayoutExample', false);
+
+    async function launchToasts(toastsContainer, buttonSelector) {
+
+      const [openToastBtn] = await waitAndGetElements(buttonSelector);
+      await openToastBtn.click();
+      await openToastBtn.click();
+
+      await wait(500);
+
+      const [container] = await waitAndGetElements(toastsContainer);
+      return container;
+    }
+
+    describe('NxToastContainer', function() {
+
+      beforeEach(async function() {
+        await getPage().setViewport({ width: 1366, height: 1000 });
+      });
+
+      it('is positioned correctly', async function() {
+        await launchToasts(toastContainer, openButton);
+        await checkFullPageScreenshot();
+      });
+
+      it('passes a11y checks', async function() {
+        await launchToasts(toastContainer, openButton);
+        a11yTest(null, true);
+      });
+    });
+  });
+
+  describe('with legacy layout', function() {
+    const {
+      checkFullPageScreenshot,
+      getPage,
+      wait,
+      waitAndGetElements,
+      a11yTest
+    } = setupBrowser('#/NxToastLegacySectionScrollingExample', false);
+
+    async function launchToasts(toastsContainer, buttonSelector) {
+
+      const [openToastBtn] = await waitAndGetElements(buttonSelector);
+      await openToastBtn.click();
+      await openToastBtn.click();
+
+      await wait(500);
+
+      const [container] = await waitAndGetElements(toastsContainer);
+      return container;
+    }
+
+    describe('NxToastContainer', function() {
+
+      beforeEach(async function() {
+        await getPage().setViewport({ width: 1366, height: 1000 });
+      });
+
+      it('is positioned correctly', async function() {
```

Comment:
We should also check that the toasts remain positioned correctly after the page or `.nx-page-main` is scrolled.

Reply:
> We should also check that the toasts remain positioned correctly after the page or .nx-page-main is scrolled.

This seems to have solved itself; many of the tests are scrolled down, probably because they had to scroll to reach the buttons.

Reply:
> > We should also check that the toasts remain positioned correctly after the page or .nx-page-main is scrolled.
> 
> This seems to have solved itself; many of the tests are scrolled down, probably because they had to scroll to reach the buttons.

Sounds good. I was just working on tests for scrolling back up to the top, but I will get rid of them 👍